### PR TITLE
Purchases: Disable cancel for pending_transfer domains

### PR DESCRIPTION
**Pull request by aidvu**
Users were able to `cancel` domains that were pending transfer through `/purchases`, although the `cancel` would always fail.

Hide the `cancel` option if the domain is `pending transfer`.

Requires D3106-code.

### CODE REVIEW

File: client/lib/purchases/index.js [hunk 1]
```diff
@@ -127,6 +131,10 @@ function isPaidWithPaypal( purchase ) {
 	return 'paypal' === purchase.payment.type;
 }
 
+function isPendingTransfer( purchase ) {
+	return true === purchase.pendingTransfer;
```
Comment (umurkontaci) [inline client/lib/purchases/index.js]: No need for an equality check here.

File: client/lib/purchases/test/index.js [hunk 2]
```diff
@@ -60,5 +61,9 @@ describe( 'index', () => {
 		it( 'should be cancelable when the purchase can have auto-renew disabled', () => {
 			expect( isCancelable( PLAN_PURCHASE ) ).to.be.true;
 		} );
+
+		it( 'should be cancelable when the purchase can have auto-renew disabled', () => {
```
Comment (umurkontaci) [inline client/lib/purchases/test/index.js]: CP mistake :) The test name needs a change

**Comment by aidvu** [comment]
@umurkontaci Updated. :)

**Comment by klimeryk** [comment]
LGTM! 🚢

**Pull request merged**